src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -148,6 +150,14 @@ func (s *fileSystemSource) Read(
 	}
 	builder := result.NewIndexBuilder(segBuilder)
 
+	// Preload info file results so they can be used to bootstrap data filesets and data migrations
+	infoFilesByNamespace := s.loadInfoFiles(namespaces)
```

Comment:
Should we cache info files even further up? Maybe somewhere at the beginning of bootstrap for all bootstrappers (if they need it)? 

Filesystem processes should not run during bootstrap so the info files should not become stale.

Reply:
Yeah, you're right on the money here. I actually looked into this when I first started working on the brief doc. You definitely can and I think it would be beneficial, but I was a little reticent as it would have been a bigger change to the bootstrapper process than I was ready to commit to at that moment. 

With the context that I have now, I'd be open to taking a crack at it, but I think I'd rather do that in a follow on PR so as to keep this one (relatively) small.

Reply:
Perhaps we should do it as a followup, agreed. But also agree we should do this.